services/app-api/utils/time/time.ts

```diff
@@ -21,18 +21,80 @@ export const convertDateUtcToEt = (date: number): string => {
 };
 
 /*
- * Calculates the period given a due date.
+ * Calculates the period given the current date.
  * The periods are defined as follows:
  *     Period 1 is from 01/01 to 06/30.
  *     Period 2 is from 07/01 to 12/31.
  */
-export const calculatePeriod = (dueDate: number, workPlan?: ReportMetadata) => {
+export const calculatePeriod = (
+  currentDate: number,
+  workPlan?: ReportMetadata
+) => {
   if (workPlan) return workPlan.reportPeriod;
-  const date = new Date(dueDate);
+  const date = new Date(currentDate);
   const period = Math.ceil((date.getMonth() + 1) / 6);
   return period.toString();
 };
 
+/**
+ * Calculates if the given year is a leap year
+ * @param year The year.
+ * @returns if the given year is a leap year.
+ */
+export const isLeapYear = (year: number) => {
+  return (year % 4 == 0 && year % 100 != 0) || year % 400 == 0;
+};
+
+/**
+ * This method returns a date in ISO format to a date in mm/dd/yyyy format.
+ * @param date The given date in ISO format
+ * @returns a date in mm/dd/yyyy format
+ */
+export const convertToFormattedDate = (date: Date) => {
+  var year = date.getFullYear();
+
+  var month = (1 + date.getMonth()).toString();
+  month = month.length > 1 ? month : "0" + month;
+
+  var day = date.getDate().toString();
+  day = day.length > 1 ? day : "0" + day;
+
+  return month + "/" + day + "/" + year;
+};
```

Comment:
Some thoughts:
1. Is `var` making a comeback? 😵 
2. Suggestion: string interpolation (combining with backticks) instead of concatenation (with pluses)
  \`${month}/${day}/${year}\` etc.

Reply:
1. Is `var` making a comeback? Not intentionally 😅 I can use `let` instead 😄
2. good suggestion!! i will revise this with the backticks